src/compiler/parser.rs

```diff
@@ -83,6 +83,12 @@ pub fn for_each_child<TNodeCallback: FnMut(Option<Rc<Node>>), TNodesCallback: Fn
         Node::TypeNode(TypeNode::ArrayTypeNode(array_type)) => {
             visit_node(&mut cb_node, Some(array_type.element_type.clone()))
         }
+        Node::TypeNode(TypeNode::UnionTypeNode(union_type)) => {
+            visit_nodes(&mut cb_node, &mut cb_nodes, Some(&union_type.types))
```

Comment:
These are a single case in the Typescript version where it casts the node to `UnionOrIntersectionTypeNode` (aka `UnionTypeNode | IntersectionTypeNode`), I could achieve something similar here by creating eg a `UnionOrIntersectionTypeNodeInterface` trait and having `UnionTypeNode` and `IntersectionTypeNode` implement it but seemed like possibly overkill (if other places keep wanting to treat a `UnionTypeNode`/`IntersectionTypeNode` symmetrically then I'll probably go ahead and do that)